lib/ui/home_page/usecases/parse_qr_code_use_case.dart

```diff
@@ -4,45 +4,52 @@ import 'package:hypha_wallet/core/crypto/seeds_esr/scan_qr_code_result_data.dart
 import 'package:hypha_wallet/core/crypto/seeds_esr/seeds_esr.dart';
 import 'package:hypha_wallet/core/error_handler/model/hypha_error.dart';
 import 'package:hypha_wallet/core/logging/log_helper.dart';
+import 'package:hypha_wallet/core/network/models/user_profile_data.dart';
 import 'package:hypha_wallet/core/network/repository/auth_repository.dart';
 import 'package:hypha_wallet/ui/architecture/interactor/base_usecase.dart';
 import 'package:hypha_wallet/ui/architecture/result/result.dart' as HResult;
+import 'package:hypha_wallet/ui/sign_transaction/usecases/create_free_transaction_use_case.dart';
 
 class ParseQRCodeUseCase extends InputUseCase<HResult.Result<ScanQrCodeResultData, HyphaError>, ParseESRLinkInput> {
   final AuthRepository _authRepository;
-  ParseQRCodeUseCase(this._authRepository);
+  final FreeTransactionUseCase _freeTransactionUseCase;
+
+  ParseQRCodeUseCase(this._authRepository, this._freeTransactionUseCase);
 
   @override
   Future<HResult.Result<ScanQrCodeResultData, HyphaError>> run(ParseESRLinkInput input) async {
     final userData = _authRepository.authDataOrCrash;
-    final qrCodeValidationResult =
-        await _validateQrCode(accountName: userData.userProfileData.accountName, scanResult: input.esrLink);
+    final user = userData.userProfileData;
+    final qrCodeValidationResult = await _validateQrCode(user: user, scanResult: input.esrLink);
 
     if (qrCodeValidationResult.isValue) {
-      return HResult.Result.value(qrCodeValidationResult.asValue!.value);
+      final data = qrCodeValidationResult.asValue!.value;
+      data.freeCpuAction = await _freeTransactionUseCase.run(user, data.transaction.network);
+      return HResult.Result.value(data);
     } else {
       return HResult.Result.error(qrCodeValidationResult.asError!.error as HyphaError);
     }
   }
-}
 
```

Comment:
the code below has not actually changed, just moved it inside the class - it was outside the class before, for whatever reason